### DISCUSSION TITLE
Interactions + Debugger/Test-Runner: Testing Library methods must be awaited

### DISCUSSION THREAD

**Discussion by sophianathanielle** [storybookjs/storybook#17664] [Help] [answered]
Hi there!
The title is referring to this excerpt in the [Interactions guide](https://storybook.js.org/docs/angular/essentials/interactions):

> While you can refer to the [Testing Library documentation](https://testing-library.com/docs/) for details on how to use it, there's an important detail that's different when using the Storybook wrapper: **method invocations must be await-ed**. It allows you to step back and forth through your interactions using the debugger.

I have Storybook set up with Angular and Jest. Currently, these Testing Library methods are not typed asynchronously, causing "Unexpected `await` of a non-Promise (non-"Thenable") value" flags/warnings all over the stories.

Is this a problem on my side? How do I configure it properly?

**Comment by yannbf** [comment, marked as answer]
Hey @sophianathanielle thanks for opening this discussion! This is a known issue and you can keep track of it here: https://github.com/storybookjs/testing-library/issues/10

Sorry for the inconvenience, but for now the only solution is to loosen the rules on your project or ts ignore them.

The rule might be coming from `@typescript-eslint/await-thenable`, and you can choose to ignore it for stories files while we find a fix for them.

**Reply by sophianathanielle** [reply]
Thanks so much! 😄
